tests/ui/macros/macro-metavar-expr-concat/syntax-errors.stderr

```diff
@@ -131,5 +131,49 @@ LL |     empty!();
    |
    = note: this error originates in the macro `empty` (in Nightly builds, run with -Z macro-backtrace for more info)
 
-error: aborting due to 18 previous errors
+error: `${concat(..)}` is not generating a valid identifier
+  --> $DIR/syntax-errors.rs:103:16
+   |
+LL |         const ${concat(_foo, $literal)}: () = ();
+   |                ^^^^^^^^^^^^^^^^^^^^^^^^
+...
+LL |     bad_literal_parameter!("\u{00BD}");
+   |     ---------------------------------- in this macro invocation
+   |
+   = note: this error originates in the macro `bad_literal_parameter` (in Nightly builds, run with -Z macro-backtrace for more info)
+
+error: `${concat(..)}` is not generating a valid identifier
+  --> $DIR/syntax-errors.rs:103:16
+   |
+LL |         const ${concat(_foo, $literal)}: () = ();
+   |                ^^^^^^^^^^^^^^^^^^^^^^^^
+...
+LL |     bad_literal_parameter!("\x41");
+   |     ------------------------------ in this macro invocation
+   |
+   = note: this error originates in the macro `bad_literal_parameter` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Comment:
I think these error messages would be improved if they print the created identifier as a note. E.g. "`_foo🤷` is not a valid identifier" can help narrow down what exactly is wrong when the exact combination isn't as obvious.

Doesn't have to be a part of this PR though.